timemachine/fe/topology.py

```diff
@@ -144,46 +144,73 @@ def parameterize_periodic_torsion(self, proper_params, improper_params):
         )
         return self._parameterize_bonded_term(guest_params, guest_potential, self.host_periodic_torsion)
 
-    def parameterize_nonbonded(self, ff_q_params, ff_q_params_intra, ff_lj_params, lamb: float):
+    def parameterize_nonbonded(self, ff_q_params, ff_q_params_intra, ff_q_params_solv, ff_lj_params, lamb: float):
         num_guest_atoms = self.guest_topology.get_num_atoms()
-        guest_params, guest_pot = self.guest_topology.parameterize_nonbonded(
-            ff_q_params, ff_q_params_intra, ff_lj_params, lamb, intramol_params=False
+        # ligand other interactions
+        guest_ixn_other_params, _ = self.guest_topology.parameterize_nonbonded(
+            ff_q_params, None, None, ff_lj_params, lamb, intramol_params=False
         )
 
-        assert guest_params.shape == (num_guest_atoms, 4)
-        assert self.host_nonbonded is not None
-        assert guest_pot.beta == self.host_nonbonded.potential.beta
-        assert guest_pot.cutoff == self.host_nonbonded.potential.cutoff
+        # ligand water interactions
+        guest_ixn_water_params, _ = self.guest_topology.parameterize_nonbonded(
+            ff_q_params_solv, None, None, ff_lj_params, lamb, intramol_params=False
+        )
+        # ligand intramolecular interactions
+        guest_intra_params, guest_intra_pot = self.guest_topology.parameterize_nonbonded_pairlist(
+            None, ff_q_params_intra, ff_lj_params, intramol_params=True
+        )
 
-        # Exclude all ligand-ligand interactions which will be computed using a pairlist instead
-        guest_exclusions, guest_scale_factors = exclude_all_ligand_ligand_ixns(self.num_host_atoms, num_guest_atoms)
+        # shift idxs because of the host
+        beta = guest_intra_pot.beta
+        cutoff = guest_intra_pot.cutoff
+        guest_intra_pot.idxs = guest_intra_pot.idxs + self.num_host_atoms
+        assert guest_ixn_water_params.shape == (num_guest_atoms, 4)
+        assert self.host_nonbonded is not None
+        assert beta == self.host_nonbonded.potential.beta
+        assert cutoff == self.host_nonbonded.potential.cutoff
 
-        hg_exclusion_idxs = np.concatenate([self.host_nonbonded.potential.exclusion_idxs, guest_exclusions])
-        hg_scale_factors = np.concatenate([self.host_nonbonded.potential.scale_factors, guest_scale_factors])
+        # Exclude all ligand-lignad interactions which will be computed using a pairlist instead
```

Comment:
```suggestion
        # Exclude all ligand-ligand interactions which will be computed using a pairlist instead
```

Reply:
Removed as comment is no longer correct (ligand atoms are excluded by the atom_idxs not the exclusion list)